### PR TITLE
pkg/lwip: overiddable settings in lwipopts.h

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the following changes:

- `TCPIP_THREAD_STACKSIZE` is made overridable. On some platforms the default stack size is too small for `lwIP`'s `tcpip_thread` so that the default stack size has to be overriden.
- Platform dependent definition of `TCPIP_THREAD_PRIO` for ESP32 is removed. This is now defined by `CFLAGS` in ESP32's Makefile.

### Testing procedure

Flash `tests/lwip` with the definition of `TCPIP_THREAD_STACKSIZE` for any ESP32 board:
```
USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"ssid\" -DESP_WIFI_PASS=\"pass\"' \
make BOARD=esp32-wroom-32 -C tests/lwip flash term
```
Use command `ps`. The stack size for thread `tcpip_thread` should be 3072 instead of the default stack size of 2048.
```
  6 | tcpip_thread         | bl mbox  _ |   2 |   3072 ( 2892) | 0x3ffb98cc | 0x3ffba260 
```

### Issues/PRs references

Figured out when testing PR #12903 for ESP32.

### CODE REVIEW

**Comment by gschorcht** [comment]
@miri64 An evaluation of you as the owner would be welcome. @benpicco or @MrKevinWeiss might be willing to test it.

File: cpu/esp32/Makefile.include [hunk 1]
```diff
@@ -129,6 +129,10 @@ ifneq (,$(filter esp_wifi_any esp_eth,$(USEMODULE)))
   CFLAGS += -DSCHED_PRIO_LEVELS=32
 endif
 
+ifneq (,$(filter lwip,$(USEMODULE)))
+  CFLAGS += -DTCPIP_THREAD_PRIO=2
+endif
```
Comment (benpicco) [inline cpu/esp32/Makefile.include]: Also move the comment from `lwipopts.h` here so there is an explanation why this is needed / where this value is coming from.

File: pkg/lwip/include/lwipopts.h [hunk 2]
```diff
@@ -141,15 +141,8 @@ extern "C" {
...
 #define NETIF_MAX_HWADDR_LEN    (GNRC_NETIF_HDR_L2ADDR_MAX_LEN)
 
+#ifndef TCPIP_THREAD_STACKSIZE
 #define TCPIP_THREAD_STACKSIZE  (THREAD_STACKSIZE_DEFAULT)
-
-#if defined(CPU_ESP32) && !defined(DOXYGEN)
-/**
- * In ESP32, the thread that is dealing with hardware interrupts of the WiFi
- * interface has a priority of 1. This thread should have a higher priority
- * than lwIP's TCP/IP thread.
```
Comment (benpicco) [inline pkg/lwip/include/lwipopts.h]: That comment is still valuable.

**Comment by miri64** [comment]
LGTM. :-)

**Comment by gschorcht** [comment]
@benpicco Squashed.

**Comment by gschorcht** [comment]
Thanks for reviewing and merging

**Pull request merged**